### PR TITLE
api to load experiments

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-1001

#### Link to staging deployment URL 

in UI repo

#### Links to any Pull Requests related to this

UI : https://github.com/biomage-ltd/ui/pull/272
API : https://github.com/biomage-ltd/api/pull/126

#### Anything else the reviewers should know about the changes here

- Created new API endpoint to fetch experiments under a project
- Created new function in `experiments` to get a list of experiments

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [X] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [X] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [X] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/api/route-services/experiment.js [hunk 1]
```diff
@@ -46,6 +46,30 @@ class ExperimentService {
     return data;
   }
 
+  async getListOfExperiments(experimentIds) {
+    const dynamodb = createDynamoDbInstance();
+
+    const params = {
+      RequestItems: {
+        [this.experimentsTableName]: {
+          Keys: [...experimentIds].map((experimentId) => convertToDynamoDbRecord({ experimentId })),
```
Comment (cosa65) [inline src/api/route-services/experiment.js:55]: Is creating a copy of experimentIds necessary here? I mean the `[...experimentIds]`

File: src/api/route-services/experiment.js [hunk 2]
```diff
@@ -46,6 +46,30 @@ class ExperimentService {
...
+        },
+      },
+    };
+
+    try {
+      const response = await dynamodb.batchGetItem(params).promise();
```
Comment (cosa65) [inline src/api/route-services/experiment.js:61]: The `batchGetItem`s have some restrictions that we have to deal with, for the previous use case of `batchGetItem` we created a ticket. We should add this use case to that ticket too.
https://biomage.atlassian.net/browse/BIOMAGE-992

File: src/specs/api.yaml [hunk 3]
```diff
@@ -668,6 +668,40 @@ paths:
...
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: ./models/api-body-schemas/Experiment.v1.yaml
+        '401':
+          description: Not Found
```
Comment (cosa65) [inline src/specs/api.yaml:693]: This should say unauthorized

**Review by cosa65 — APPROVED** [review]
Approved, the comments and tests will be addressed in a new PR to unblock other PRs depending on this

File: src/api/route-services/experiment.js (hunk 1 above)
Comment (aerlaut) [reply src/api/route-services/experiment.js:55]: Hmm... yea, they don't have to be copied. Fixed.

File: src/specs/api.yaml (hunk 3 above)
Comment (aerlaut) [reply src/specs/api.yaml:693]: Fixed

**Pull request merged**